src/utils/misc.js

```diff
@@ -32,3 +32,18 @@ export function groupItemsById<T: { +id: number }>(items: T[]): { [id: number]:
 }
 
 export const isValidEmailFormat = (email: string = ''): boolean => /\S+@\S+\.\S+/.test(email);
+
+export const initialsForGroupIcon = (name: string): string => {
+  const Name = name.split(', ');
+  let x = '';
+  for (let i = 0; i < Name.length; i++) {
+    if (i < 4) {
+      x += Name[i][0].toUpperCase();
+    } else {
+      x = x.slice(0, -1);
+      x += '+';
+      break;
+    }
+  }
```

Comment:
This can be simplified significantly by testing `Name.length` once, and using `.map()` rather than an explicit loop.

Extracting a single character from a string in JavaScript with `string[n]` is almost invariably always a bug; this should use `.codePointAt(0)` instead. (Its absence is a bug in `initialsFromString`!) If a user has a name beginning with a character not in the Basic Multilingual Plane (e.g., an emoji), `string[0]` will break up a surrogate pair, which will cause many problems.

Reply:
**Erratum:** `.codePointAt(0)` returns the numeric value of the first code point. There are several options to extract the first character; perhaps the most straightforward is to check for a high surrogate, and if one is found, to take the following low surrogate as well.